pkg/console/service/service_test.go

```diff
@@ -0,0 +1,236 @@
+				return nil, errors.NewNotFound(v1.Resource("virtualmachines"), testName)
+			},
+		).AnyTimes()
+
+		virtClient.EXPECT().VirtualMachine(testNamespace).Return(vmInterface).AnyTimes()
+
+		testService = &service{
+			kubevirtClient: virtClient,
+		}
+
+		request = restful.NewRequest(&http.Request{
+			Header: make(http.Header),
+		})
+		request.PathParameters()["namespace"] = testNamespace
+		request.PathParameters()["name"] = testName
+
+		recorder = httptest.NewRecorder()
+		response = restful.NewResponse(recorder)
+		response.SetRequestAccepts(restful.MIME_JSON)
+	})
+
+	Context("TokenHandler", func() {
+		const (
+			testToken           = "test-token-value"
+			authorizationHeader = "Authorization"
+		)
+
+		BeforeEach(func() {
+			const validToken = "test-auth-token"
+			apiClient.Fake.PrependReactor("create", "tokenreviews", func(action k8stesting.Action) (bool, runtime.Object, error) {
+				createAction := action.(k8stesting.CreateAction)
+				tokenReview := createAction.GetObject().(*authnv1.TokenReview)
+				tokenReview.Status.Authenticated = true
+				return true, tokenReview, nil
+			})
+
+			apiClient.Fake.PrependReactor("create", "subjectaccessreviews", func(action k8stesting.Action) (bool, runtime.Object, error) {
+				createAction := action.(k8stesting.CreateAction)
+				sar := createAction.GetObject().(*authzv1.SubjectAccessReview)
+				sar.Status.Allowed = true
+				return true, sar, nil
+			})
+
+			apiClient.Fake.PrependReactor("create", "serviceaccounts/token", func(action k8stesting.Action) (bool, runtime.Object, error) {
+				createAction := action.(k8stesting.CreateAction)
+				tokenRequest := createAction.GetObject().(*authnv1.TokenRequest)
+				tokenRequest.Status.Token = testToken
+				return true, tokenRequest, nil
+			})
+
+			request.Request.Header.Set(authorizationHeader, "Bearer "+validToken)
+			// Using a dummy URL, so tests don't panic
+			requestUrl, err := url.Parse("example.org/api")
+			Expect(err).ToNot(HaveOccurred())
+			request.Request.URL = requestUrl
+		})
+
+		It("should fail if namespace is empty", func() {
+			delete(request.PathParameters(), "namespace")
+
+			testService.TokenHandler(request, response)
+
+			Expect(recorder.Code).To(Equal(http.StatusBadRequest))
+			Expect(recorder.Body.String()).To(ContainSubstring("namespace and name parameters are required"))
+		})
+
+		It("should fail if name is empty", func() {
+			delete(request.PathParameters(), "name")
+
+			testService.TokenHandler(request, response)
+
+			Expect(recorder.Code).To(Equal(http.StatusBadRequest))
+			Expect(recorder.Body.String()).To(ContainSubstring("namespace and name parameters are required"))
+		})
+
+		It("should fail if no Authorization header is provided", func() {
+			request.Request.Header.Del(authorizationHeader)
+
+			testService.TokenHandler(request, response)
+
+			Expect(recorder.Code).To(Equal(http.StatusUnauthorized))
+			Expect(recorder.Body.String()).To(Equal("authenticating token cannot be empty"))
```

Comment:
IMO it would be better to just return 401 without any further reasons. Do not provide more information than necessary.